internal/service/identitystore/group_membership_test.go

```diff
@@ -0,0 +1,325 @@
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckGroupMembershipDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGroupMembershipConfig_basic(rName1, rName2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGroupMembershipExists(resourceName, &groupMembership),
+					resource.TestCheckResourceAttrPair(resourceName, "member_id", groupResourceName, "user_id"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccGroupMembershipConfig_basic(rName1, rName3),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGroupMembershipExists(resourceName, &groupMembership),
+					resource.TestCheckResourceAttrPair(resourceName, "member_id", groupResourceName, "user_id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGroupMembershipDestroy(s *terraform.State) error {
+	conn := acctest.Provider.Meta().(*conns.AWSClient).IdentityStoreConn
+	ctx := context.Background()
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_identitystore_group_membership" {
+			continue
+		}
+
+		_, err := conn.DescribeGroupMembership(ctx, &identitystore.DescribeGroupMembershipInput{
+			IdentityStoreId: aws.String(rs.Primary.Attributes["identity_store_id"]),
+			MembershipId:    aws.String(rs.Primary.Attributes["membership_id"]),
+		})
+		if err != nil {
+			var nfe *types.ResourceNotFoundException
+			if errors.As(err, &nfe) {
+				return nil
+			}
+			return err
+		}
+
+		return create.Error(names.IdentityStore, create.ErrActionCheckingDestroyed, tfidentitystore.ResNameGroupMembership, rs.Primary.ID, errors.New("not destroyed"))
+	}
+
+	return nil
+}
+
+func testAccCheckGroupMembershipExists(name string, groupMembership *identitystore.DescribeGroupMembershipOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return create.Error(names.IdentityStore, create.ErrActionCheckingExistence, tfidentitystore.ResNameGroupMembership, name, errors.New("not found"))
+		}
+
+		if rs.Primary.ID == "" {
+			return create.Error(names.IdentityStore, create.ErrActionCheckingExistence, tfidentitystore.ResNameGroupMembership, name, errors.New("not set"))
+		}
+
+		conn := acctest.Provider.Meta().(*conns.AWSClient).IdentityStoreConn
+		ctx := context.Background()
+		resp, err := conn.DescribeGroupMembership(ctx, &identitystore.DescribeGroupMembershipInput{
+			IdentityStoreId: aws.String(rs.Primary.Attributes["identity_store_id"]),
+			MembershipId:    aws.String(rs.Primary.Attributes["membership_id"]),
+		})
+
+		if err != nil {
+			return create.Error(names.IdentityStore, create.ErrActionCheckingExistence, tfidentitystore.ResNameGroupMembership, rs.Primary.ID, err)
+		}
+
+		*groupMembership = *resp
+
+		return nil
+	}
+}
+
+func testAccPreCheck(t *testing.T) {
```

Comment:
I believe in main there's already a testAccPreCheck function, could you rebase on main and remove this duplicate code?

Reply:
Done